Backendless/Data/DictionaryDrivenDataStore.cs

```diff
@@ -714,6 +658,47 @@ public async Task<int> DeleteRelationAsync( Dictionary<string, object> parent, s
       Backendless.Data.DeleteRelation<Dictionary<string, object>>( tableName, parent, columnName, children, callback );
     }
 
-  #endregion
+    #endregion
+
+  #region CREATE_ARGS
+    private Object[] CreateArgs( DataQueryBuilder qb )
+    {
+      return SubArgsCreator<Object>( qb.GetRelated(), qb.GetRelationsDepth() );
+    }
+
+    private Object[] CreateArgs( String id, IList<String> relations, int? relationsDepth )
+    {
+      if( id == null )
+        throw new ArgumentNullException( ExceptionMessage.NULL_ID );
+
+      return SubArgsCreator<String>( relations, relationsDepth, id );
+    }
+
+    private Object[] CreateArgs( Dictionary<String, Object> entity, IList<String> relations, int? relationsDepth )
+    {
+      if( entity == null )
+        throw new ArgumentNullException( ExceptionMessage.NULL_ENTITY );
+
+      return SubArgsCreator<Dictionary<String, Object>>( relations, relationsDepth, entity );
+    }
+
+    private Object[] SubArgsCreator<T>( IList<String> relations, int? Depth, T obj = null ) where T : class

```

Comment:
"where T : class". Is T really a class? The way the method is used is "obj" can be either a Dictionary or String, so I am wondering why you need to specify that "T : class"?